src/prpy/perception/rock_module.py

```diff
@@ -0,0 +1,140 @@
+from base import PerceptionModule, PerceptionMethod
+
+logger = logging.getLogger(__name__)
+
+class RockModule(PerceptionModule):
+
+    def __init__(self, kinbody_path, detection_frame, destination_frame):
+        super(RockModule, self).__init__()
+        self.kinbody_path = kinbody_path
+        self.detection_frame = detection_frame
+        self.destination_frame = destination_frame
+
+    def __str__(self):
+        return 'RockModule'
+
+    def _GetDetector(self, env, kinbody_path, detection_frame, destination_frame):
+        """
+        @param env The OpenRAVE environment
+        @param kinbody_path The path to use to map kinbodies to poses 
+           (if None defaults to the path used to initialize the module)
+        @param detection_frame The camera frame
+           (if None defaults to the frame used to initialize the module)
+        @param destination_frame The frame to place detected objects into
+           (if None defaults to the frame used to intialize the module)
+        """
+
+        if kinbody_path is None:
+            kinbody_path = self.kinbody_path
+
+        if detection_frame is None:
+            detection_frame = self.detection_frame
+            
+        if destination_frame is None:
+            destination_frame = self.destination_frame
+
+        from rock import RockDetector
+        rock_detector = RockDetector(
+            env,
+            kinbody_path,
+            detection_frame=detection_frame,
+            world_frame=destination_frame)
+        return rock_detector
+
+    @PerceptionMethod
+    def DetectObject(self, robot, obj_name,
+                     kinbody_path=None,
+                     detection_frame=None,
+                     destination_frame=None):
+        """
+        @param obj_name The name of the object to search for
+        @param kinbody_path The path to use to map kinbodies to poses 
+           (if None defaults to the path used to initialize the module)
+        @param detection_frame The camera frame
+           (if None defaults to the frame used to initialize the module)
+        @param destination_frame The frame to place detected objects into
+           (if None defaults to the frame used to intialize the module)
+        @return A list of detected kinbodies
+        """
+        # Initial the detector
+        rock_detector = self._GetDetector(robot.GetEnv(),
+                                          kinbody_path,
+                                          detection_frame,
+                                          destination_frame)
+    
+        # Use the detector to find object poses
+        poses = rock_detector.SearchTabletop(obj_name)
+    
+        # Use the poses to generate new kinbodies
+        kinbodies = rock_detector.KinBodiesFromPoses(obj_names, poses)
+
+        return kinbodies
+
+    @PerceptionMethod
+    def DetectObjectNearPose(self, robot, obj_name, pose, 
+                             search_extents=[-0.1, -0.1, 0.],
```

Comment:
We should [*never* set default values to mutable objects](https://pythonconquerstheuniverse.wordpress.com/2012/02/15/mutable-default-arguments/). This should be changed to `None` and an `if is None` check in the function body.